File: game/game_data.py
from dataclasses import dataclass


@dataclass
class GameData:
    country: str
    audio_port: int

    forest_score: int
    forest_lives: int
    forest_parallax_pos: int
    forest_sacks: list
    forest_obstacles: list
    forest_leaves: list
    forest_controls_inverted: bool

    # Scoreboard tracking
    forest_normal_sacks_collected: int = 0
    forest_golden_sacks_collected: int = 0
    forest_reached_end: bool = False
    forest_obstacles_hit: list = None  # List of obstacle types that caused life loss

    cave_selected_rope: int = 0
    cave_win_type: int = 0

    forest_bg_atmosphere_id: int = None
    cave_score_counter_id: int = None
    cave_fanfare_id: int = None
    cave_bg_music_id: int = None
    cave_stemning_id: int = None
    scoreboard_score_counter_id: int = None

File: game-web/src/gameData.ts
export class GameData {
  country: string;
  audioPort: number;  // kept for reference; not used (Web Audio replaces UDP)

  // Forest game state
  forestScore: number;
  forestLives: number;
  forestParallaxPos: number;
  forestSacks: unknown[];
  forestObstacles: unknown[];
  forestLeaves: unknown[];
  forestControlsInverted: boolean;

  // Forest scoring
  forestNormalSacksCollected = 0;
  forestGoldenSacksCollected = 0;
  forestReachedEnd = false;
  forestObstaclesHit: string[] = [];

  // Cave game state
  caveSelectedRope = 0;
  caveWinType = 0;

  // Audio instance IDs (Web Audio AudioBufferSourceNode wrappers)
  forestBgAtmosphereId: number | null = null;
  caveScoreCounterId: number | null = null;
  caveFanfareId: number | null = null;
  caveBgMusicId: number | null = null;
  caveStemningId: number | null = null;
  scoreboardScoreCounterId: number | null = null;

  constructor(
    country: string,
    audioPort: number,
    forestScore: number,
    forestLives: number,
    forestParallaxPos: number,
    forestSacks: unknown[],
    forestObstacles: unknown[],
    forestLeaves: unknown[],
    forestControlsInverted: boolean,
    forestNormalSacksCollected: number,
    forestGoldenSacksCollected: number,
  ) {
    this.country = country;
    this.audioPort = audioPort;
    this.forestScore = forestScore;
    this.forestLives = forestLives;
    this.forestParallaxPos = forestParallaxPos;
    this.forestSacks = forestSacks;
    this.forestObstacles = forestObstacles;
    this.forestLeaves = forestLeaves;
    this.forestControlsInverted = forestControlsInverted;
    this.forestNormalSacksCollected = forestNormalSacksCollected;
    this.forestGoldenSacksCollected = forestGoldenSacksCollected;
  }
}
